### ISSUE TITLE
feature: make the float window width adaptive

### ISSUE THREAD

**Issue by OnlyWick** [LazyVim/LazyVim#4962]
### Did you check the docs?

- [x] I have read all the LazyVim docs
- [x] This is not a request for a new or existing extra (see above)

### Is your feature request related to a problem? Please describe.

If the terminal window is too small, the float window may occupy the entire window.

![Image](https://github.com/user-attachments/assets/51ff4a8d-ec20-493d-9938-52f8affbfaae)

### Describe the solution you'd like

Is there a way to make a floating window adapt to the terminal window size, for example, occupying 80% of the terminal window width?

![Image](https://github.com/user-attachments/assets/4a6f7128-ca08-4a28-8a36-a782d7007378)

### Describe alternatives you've considered

Float windows should have a certain level of width adaptability, such as minimum and maximum widths.

**Comment by dpetka2001** [comment]
That comes from `noice.nvim`. Read its docs if there's an option to configure it or ask for a feature at the corresponding repo.

**Comment by OnlyWick** [comment]
@dpetka2001 Thank you for the reminder. I have previously read the relevant issue and confirmed that this is the behavior of noice.nvim. Personally, I think this adaptive width feature should be built into LazyVim to achieve out-of-the-box(The float window of cmp often takes up the entire screen, which makes me feel uncomfortable).

But, I just read the documentation for [nui.nvim](https://github.com/MunifTanjim/nui.nvim/tree/main/lua/nui/popup#size), and it seems that there is no configuration option for setting the maximum width.

I will temporarily close this issue for now.

**Comment by dpetka2001** [comment]
LazyVim just uses the plugins as they come. It's up to the plugins what features they provide (such as configurable window size).

**Comment by OnlyWick** [comment]
@dpetka2001 I didn't find any documentation on this in `nui.nvim`, so I thought about adding a feature for setting maximum width and height. I started debugging `nui.nvim` and found that folke had already made changes to nui.nvim to accommodate `noice.nvim` two years ago.

And then I discovered that [this configuration](https://github.com/folke/noice.nvim/blob/72e21399c7b292a2470d9230bf63cad5d6375140/lua/noice/config/views.lua#L131) would be passed into `nui.nvim`.

So if we want to set the maximum display width for popups, we can make modifications to `noice.nvim`.(`The max_width: maximum width of the message` that I saw in noice.nvim doesn't seem to be related to setting the maximum width for popups.`).

Another feature I feel should be added is that the popup window should ideally not overlap with the neo-tree.
![Image](https://github.com/user-attachments/assets/823d43d9-cd5d-4714-ba57-771792cddeba)
